bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageHelperTest.java

```diff
@@ -0,0 +1,102 @@
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.models.v2;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.testing.mock.sling.servlet.MockSlingHttpServletRequest;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.adobe.cq.wcm.core.components.context.CoreComponentTestContext;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.designer.Style;
+import io.wcm.testing.mock.aem.junit.AemContext;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class PageHelperTest {
+
+    @ClassRule
+    public static final AemContext CONTEXT = CoreComponentTestContext.createContext();
+
+    @Test
+    public void testGetModelUrl() {
+        assertNull(PageHelpers.getModelUrl(""));
+        assertEquals("some/path.model.json", PageHelpers.getModelUrl("some/path"));
+        assertEquals("path/with/some.model.json", PageHelpers.getModelUrl("path/with/some.more.selectors.html"));
+    }
+
+    @Test
+    public void testGetHierarchyServletRequest() {
+        MockSlingHttpServletRequest mockedRequest = new MockSlingHttpServletRequest(CONTEXT.resourceResolver(), CONTEXT.bundleContext());
+        Page mockedPage = mock(Page.class);
+
+        SlingHttpServletRequest hierarchyServletRequest = PageHelpers.getHierarchyServletRequest(mockedRequest, mockedPage);
+
+        assertEquals(mockedPage, hierarchyServletRequest.getAttribute("currentPage"));
+        assertNotNull(hierarchyServletRequest.getAttribute("com.day.cq.wcm.componentcontext"));
+    }
+
+    @Test
+    public void testGetPageTreeTraversalDepth() {
+        // no style
+        assertEquals(0, PageHelpers.getPageTreeTraversalDepth(null));
+
+        // structureDepth set to 42
+        Style mockedStyle = mock(Style.class);
+        int answer = 42;
+        when(mockedStyle.get(eq("structureDepth"), any())).thenReturn(answer);
+
+        assertEquals(answer, PageHelpers.getPageTreeTraversalDepth(mockedStyle));
+    }
+
+    @Test
+    public void testGetStructurePatterns() {
+        final String STRUCTURE_PATTERNS_PN = "structurePatterns";
+
+        SlingHttpServletRequest mockedRequest = mock(SlingHttpServletRequest.class);
+        RequestParameter mockedRequestParameter = mock(RequestParameter.class);
+        when(mockedRequest.getRequestParameter(eq(STRUCTURE_PATTERNS_PN.toLowerCase()))).thenReturn(mockedRequestParameter);
```

Comment:
We may want to refactor the constants names with ATTR and PN from being suffixed to be prefixed as it is more in line with convention. e.g. `PN_STRUCTURE_PATTERNS`